### DISCUSSION TITLE
Passing secrets into a re usable workflow (caller and called workflow in the same repo)

### DISCUSSION THREAD

**Discussion by WernerMairl** [community/community#51729] [Actions]
### Select Topic Area

Question

### Body

Hi,
not sure if this is bug or i miss something....any idea and help is welcome!

Usecase: 

the "called" workflow is doing a "dotnet restore" consuming Github Packages (NuGet) and it needs a Token.
Same Token is working in a lot of other workflows. This is my first try to setup re-usable workflows.
Both workflows are located in the same repo to avoid (bug isolation) the special security requirements for reusing workflows from other repos!

The workflow call basically looks like:
<img width="304" alt="image" src="https://user-images.githubusercontent.com/10084630/229337440-3165a336-d3b8-47df-978f-9917264177b0.png">

everything is **working** if I call it with the following (security breaking) syntax:
<img width="438" alt="image" src="https://user-images.githubusercontent.com/10084630/229337624-3bf6f302-35b2-4fb6-b3d7-57e1f9cb5253.png">

Using the following syntax, i get Http 401 (not authenticated)

<img width="425" alt="image" src="https://user-images.githubusercontent.com/10084630/229337727-0709e7e8-97ce-4ff2-9217-d20304a5bb2a.png">

by the way the env variable is defined (security breaking - but I'm on investigating...)
<img width="400" alt="image" src="https://user-images.githubusercontent.com/10084630/229337808-29243b32-6f89-459b-a896-860aa890a9a5.png">

using the "NUGET_CLIENT_TOKEN variable directly inside the caller Workflow also works 
using "secrets: inherit" also allows me to implement a workaround...

after 2 days of investigations (isolating effects) I must assume a bug or misunderstanding in how the "secrets" value is resolved on the CALLER side.

because: 
- passing the secret as hard coded literal works
- passing the secret as expression  (syntax ${{ }}) or env variable (Syntax ${}) doesn't work.

any idea ?
any known issue that i my not have seen ?

BR
Werner

**Comment by vitalyk-multinarity** [comment]
>passing the secret as expression (syntax ${{ }}) or env variable (Syntax ${}) doesn't work.

Weird - it works for me as 

```
    secrets:
      GCP_SA_KEY: ${{ secrets.GCP_SA_KEY }}
```

**Comment by WernerMairl** [comment]
Now I have it up and running, but I don't know exactly why (have to go back step by step to isolate the exact issue)


Changed:
- secret names in UPPERCASE and underscores (not lowercase, no minus)
- avoided putting secrets into env variables before using them again... always using expressions
